### DISCUSSION TITLE
hrp3をhrp2にダウングレードに関する質問

### DISCUSSION THREAD

**Discussion by sanzomatsumoto** [toppers/users-forum#129] [ET-Robocon]
お忙しいところすみません。
　現在、ETロボコンへの参加に際し、TOPPERS/HRP3カーネルをベースとしたソフトウェアを使用しています。
　最近、実機においてエラーが発生し、予期せぬ動きをしてしまいます。そのため、調査の結果、hrp2にダウングレードすると問題が解決するのではないかという結論に至りました。
　そこで質問なのですが、hrp3からhrp2にダウングレードが可能なのか、また可能であればその手順をご教授いただけると幸いです。
　お忙しいところ恐縮ですが、回答の程よろしくお願いいたします。

**Comment by YutakaMatsubara** [comment]
@sanzomatsumoto さま

EV3RTのカーネルは，バージョン1.0以降にHRP3カーネルになりましたので，β7-3をダウンロードしてお試し頂くのが早そうです。
環境構築手順は，基本的には同じです。
https://dev.toppers.jp/trac_user/ev3pf/wiki/Download

> 実機においてエラーが発生し、予期せぬ動きをしてしまいます。そのため、調査の結果、hrp2にダウングレードすると問題が解決するのではないか

すでにご存知かも知れませんが，カーネルのバージョン違いに関して，アプリケーション開発で気をつけて頂きたい点を以下にまとめてあります。HRP3向けのアプリケーションをHRP2で試す際には，念のため，ご確認頂ければと思います。

https://dev.toppers.jp/trac_user/ev3pf/wiki/FAQ#QEV3RTβ7-3とv1.0でカーネルがHRP2からHRP3に変わったようなのですがどのような違いがありますか

また，「予期せぬ動き」に関して，別途詳しくお教え頂ければ何か原因が分かるかも知れません。
